src/extract.js

```diff
@@ -81,20 +81,34 @@ function preprocessTemplate(data, type) {
   case 'jade':
   case 'pug':
     // Add empty require function to the context to avoid errors with webpack require inside pug
-    templateData = pug.render(data, {filename: 'source.html', pretty: true, require: function() {}});
+    templateData = pug.render(data, {filename: 'source.html', pretty: true, require: function() {}}).trim();
     break;
   case 'vue':
-    const vueFile = vueCompiler.parse({ compiler, source: data, needMap: false});
-    if (!vueFile.template) return '';  // return an empty string
-    templateData = vueFile.template.content;
-    if (vueFile.template.lang) {
-      return preprocessTemplate(templateData, vueFile.template.lang);
+    const $ = cheerio.load(templateData, {
+      xmlMode: true,
+      decodeEntities: false,
+      withStartIndices: true,
+    });
+
+    templateData = $('template').map(function() {
+      const t = $.html($(this));
+      const vueFile = vueCompiler.parse({ compiler, source: t, needMap: false});
```

Comment:
wouldn't it be possible to get the `lang` attribute directly on `this` with cheerio ?
so that we don't need to call the vueCompiler.
```javascript
const $ = cheerio.load("<tamplate lang='abc'></template><template lang='scss'></template>")
$('template').map(function () { return $(this).attr('lang') }).get()
// [ 'abc', 'scss' ]
```
just a suggestion.

Reply:
I followed the previous code. But notice that I'm using the parsed data, and I believe that the Vue parser is not just getting the `<template>` tag, but doing something. But if your suggestion works, much better.

Reply:
Actually I tried and the vue parser is not necessary. Although it's not the language that's returned, but the `<template>` contents.